### PR TITLE
Extending recovery network input in replication intent API

### PR DESCRIPTION
**Pull request by om-nishant**
Replication Intent API which is in preview phase supported taking existing virtual network input as an input. With these changes, it would be possible to take inputs to create a new recovery network through the replication intent API.
The changes extend the RecoveryNetwork property type to include new type for network creation.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. End of December 2020.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. End of December 2020.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by varunkumta** [comment]
I feel this is technically a breaking change, since you are adding properties to an existing api model to support the new vnet creation scenario .. I am manually adding the BreakingChangeReviewRequired label - please follow the breaking changes policy and request an approval

**Comment by varunkumta** [comment]
@leni-msft - I think the CI tooling needs to be updated so it can catch such breaking changes

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 1]
```diff
@@ -15724,6 +15724,26 @@
...
+    "NewRecoveryVirtualNetwork": {
+      "description": "Recovery virtual network input to create new virtual network from given source network.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/RecoveryVirtualNetworkCustomDetails"
+        }
+      ],
+      "properties": {
+        "recoveryVirtualNetworkResourceGroupName": {
```
Comment (varunkumta) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:15736]: I assume you will be creating this vnet via S2S call from your RP to Network RP - which means
- your service role definition needs to be updated by NRP team with Microsoft.Network/virtualNetworks/write permission (if you don't already have it)
- the subscription in which you create this vnet must have your RP registered - if the subscription you create the vnet in is the target VM's subscription then I assume you assert that the VM's sub == the sub which contains the recovery service's vault

Comment (om-nishant) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:15736]: Service already has option for network creation through different flow. This update would merely plug into existing workflow. So all the necessary permissions are already in place.

**Comment by om-nishant** [comment]
> BreakingChangeReviewRequired

The API in question is technically not yet GA. It is only being used by internal clients. @vigaur to confirm.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json (hunk 1 above)
Comment (varunkumta) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:15736]: thanks for confirming

**Comment by varunkumta** [comment]
> > BreakingChangeReviewRequired
> 
> The API in question is technically not yet GA. It is only being used by internal clients. @vigaur Vinayak Gaur FTE to confirm.

ok, I checked the ARM logs and in last 30 days only 2 subscriptions have made calls to the PUT vaults/replicationProtectionIntents API so I guess these are the internal/testing subscriptions you are using

**Review by varunkumta — APPROVED** [review]
approved - the sdk assignee @leni-msft will merge your PR once they have reviewed it

**Pull request merged**